### ISSUE TITLE
Places: fields parameter serialization

### ISSUE THREAD

**Issue by eglocc** [googlemaps/google-maps-services-js#314]
### Problem:

I noticed an inconsistency between the Places API docs and Node.js client library related to fields parameter serialization in findPlaceFromText function. Docs shows following examples and `fields` values are separated with commas.

<img width="935" alt="Place_Search_ _ _Places_API_ _ _Google_Developers" src="https://user-images.githubusercontent.com/27420585/74077043-9ca0f780-4a2d-11ea-9907-c22c3bd7dbd9.png">

However, this code is not functioning.

```javascript
require('dotenv').config();
const maps = new require('@googlemaps/google-maps-services-js');
const client = new maps.Client({});

client.findPlaceFromText({
  params: {
    input: 'SushiCo',
    inputtype: 'textquery',
    language: 'tr',
    fields: [
      "place_id",
      "name",
      "formatted_address"
    ],
    key: process.env.API_KEY
  },
  timeout: 1000
}).then(response => {
  console.log(JSON.stringify(response.data, null, 2));
}).catch(error => {
  console.error(error);
});
```

And it generates this error:

```json
{
  "candidates": [],
  "error_message": "Error while parsing 'fields' parameter: Unexpected character '|'. ",
  "status": "INVALID_REQUEST"
}
```

### Steps to reproduce:
Either using `|` separated fields parameter in a url like this `https://maps.googleapis.com/maps/api/place/findplacefromtext/json?input=google&inputtype=textquery&fields=place_id|name|formatted_address&key=[API KEY]` or using Node.js client library for this endpoint and not omitting `fields` parameter.

### Cause:

Below function which is located in `src/serialize.ts` serializes array parameters with `|` seperator which is required in other endpoints. However, `maps/api/place/findplacefromtext/json` endpoint needs comma separated values.

```javascript
export function serializer(
  format: serializerFormat,
  queryStringOptions: object = {
    arrayFormat: "separator",
    arrayFormatSeparator: separator
  }
) {
  return (params: { [key: string]: any }) => {
    Object.keys(format).forEach((key: string) => {
      if (key in params) {
        params[key] = format[key](params[key]);
      }
    });
    return qs(params, queryStringOptions);
  };
}
```

### Solution offer for Node.js client library:

This line in `src/places/findplacefromtext.ts` uses the default query string options. 

```javascript
export const defaultParamsSerializer = serializer({});
```

I am not really sure my suggestion might create any side-effects but I think overriding the default query string option like this would solve the problem. 

```javascript
export const defaultParamsSerializer = serializer({}, { arrayFormat: "comma" });
```

### What to do until it is fixed?

I think this issue will be closed really soon but I still share my solution which overrides paramsSerializer like below. `serializer` function is taken by the source code and modified.

You can also try the old version published to [@google/maps](https://www.npmjs.com/package/@google/maps).

```javascript
require('dotenv').config();

const qs = require('query-string');
const maps = new require('@googlemaps/google-maps-services-js');
const client = new maps.Client({});

client.findPlaceFromText({
  params: {
    input: 'SushiCo',
    inputtype: 'textquery',
    language: 'tr',
    fields: [
      "place_id",
      "name",
      "formatted_address"
    ],
    key: process.env.API_KEY
  },
  paramsSerializer: serializer({}),
  timeout: 1000
}).then(response => {
  console.log(JSON.stringify(response.data, null, 2));
}).catch(error => {
  console.error(error);
});

function serializer(format) {
  return (params) => {
    Object.keys(format).forEach(key => {
      if (key in params) {
        params[key] = format[key](params[key]);
      }
    });
    return qs.stringify(params, {
      arrayFormat: "comma"
    });
  };
}
```

### Notes:

- No test case with fields parameter exists currently in `src/places/findplacefromtext.test.ts`.
- This bug might occur in other library functions as well. I did not check others.
- If you are planning any changes about this on the server side (like separating values with `|`), please update the docs after that.

**Comment by jpoehnelt** [comment]
Thanks for reporting. I'll take a closer look on Monday. Another workaround could be to pass the fields already concatenated.

**Comment by jpoehnelt** [comment]
This is published in [v2.0.2](https://www.npmjs.com/package/@googlemaps/google-maps-services-js/v/2.0.2).
